lxd/storage/backend_lxd.go

```diff
@@ -1439,6 +1439,14 @@ func (b *lxdBackend) RefreshCustomVolume(projectName string, srcProjectName stri
 
 		targetSnapshots := make([]drivers.Volume, 0, len(targetSnaps))
 		for _, targetSnap := range targetSnaps {
+			// Quick check.
+			err := instancetype.ValidName(targetSnap.Name, true)
+			if err != nil {
+				// This should not happen as the target snapshots list is coming from the database.
```

Comment:
indeed, why is it needed?

I dont like the fact that it doesnt error here but instead logs only a debug line and then silently stops considering the snapshot.

We should be validating input before we do anything, not midway through the operation.

Reply:
I've changed it to error out now.

The sole reason for this commit is to please CodeQL that apparently lost track that the snapshot list came from the DB.

Reply:
hrm, perhaps we should mark that alert as invalid then rather than add low level internal validation for something that has already been validated.